crates/kubelet/src/store/mod.rs

```diff
@@ -72,3 +124,82 @@ pub trait Store {
             .collect()
     }
 }
+
+/// A `Store` implementation which obtains module data from remote registries
+/// but caches it in local storage.
+pub struct CachingStore<S: Storer, C: Client> {
+    storer: Arc<RwLock<S>>,
+    client: Arc<Mutex<C>>,
+}
+
+impl<S: Storer, C: Client> CachingStore<S, C> {
+    async fn pull(&self, image_ref: &Reference) -> anyhow::Result<()> {
+        debug!("Pulling image ref '{:?}' from registry", image_ref);
+        let image_data = self.client.lock().await.pull(image_ref).await?;
```

Comment:
Does the client need a lock? Just thinking through it and if all it is doing is pulling an image, I don't think there is any need to lock it, but I could be way off here

Reply:
Not sure.  The lock was there before so I cargo-culted it.

Reply:
Very well, let's keep it